addon/components/freestyle/bootstrap-pagination-nav.js

```diff
@@ -3,9 +3,24 @@ import layout from '../../templates/components/freestyle/bootstrap-pagination-na
 
 export default Ember.Component.extend({
   layout: layout,
+  numberOfElements: 100,
   pageNumber: 1,
   pageSize: 5,
-  totalPages: Ember.computed('pageSize', function() {
-    return (100 / this.get('pageSize'));
+
+  totalPages: Ember.computed('pageSize', 'numberOfElements', function() {
+    if(this.get('pageSize') && this.get('pageSize') > 0) {
+      return Math.ceil(this.get('numberOfElements') / this.get('pageSize'));
+    } else {
+      return this.get('numberOfElements');
+    }
+  }),
+  elements: Ember.computed('pageSize', 'numberOfElements', 'pageNumber', function() {
```

Comment:
I don't see `numberOfElements` used here, should be able to remove the watch on that property.